src/main/java/seedu/address/logic/commands/RenameTagCommand.java

```diff
@@ -24,6 +47,42 @@ public RenameTagCommand(String oldTag, String newTag) {
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
-        return new CommandResult(oldTag + " tag is renamed to " + newTag);
+        ObservableList<Person> persons = model.getPersonList();
+        boolean found = false;
+        for (int i = 0; i < persons.size(); i++) {
+            Person currPerson = persons.get(i);
+            Set<Tag> tags = currPerson.getTags();
+            Iterator<Tag> iterator = tags.iterator();
+            while (iterator.hasNext()) {
+                Tag tag = iterator.next();
+                if (tag.getTagName().equals(oldTag)) {
+                    tag.updateTagName(newTag);
+                    found = true;
+                }
+            }
+        }
+        model.updateFilteredPersonList(PREDICATE_DO_NOT_SHOW_ALL_PERSONS);
```

Comment:
maybe this line is redundant? since we will be showing all the contacts after

Reply:
idrk how this works but initially when i didnt include this line and i tried it out, when i call the renametag command the second time onwards the list of contacts doesnt update immediately on the app until i like click around which was kinda weird. but w this line the list did update immediately so im guessing this kinda refreshes the system?